code/_helpers/game.dm

```diff
@@ -245,6 +245,39 @@
 					. += M
 	return .
 
+/proc/get_mobs_and_objs_in_view_fast(var/turf/T, var/range, var/list/mobs, var/list/objs, var/checkghosts = GHOSTS_ALL_HEAR)
+
+	var/list/hear = list()
+	DVIEW(hear, range, T, INVISIBILITY_MAXIMUM)
+	var/list/hearturfs = list()
+
+	for(var/atom/movable/AM in hear)
+		if(ismob(AM))
+			mobs += AM
+			hearturfs += AM.locs[1]
+		else if(isobj(AM))
+			objs += AM
+			hearturfs += AM.locs[1]
+
+
+	for(var/mob/M in player_list)

```

Comment:
Nanako's correct. However, this is why there's a `for (var/m in player_list)` loop to assist the DVIEW loop.

As discussed, the only things this entire proc will not account for is clientless holder mobs. And as discussed, this is a non-issue.